### PR TITLE
Hail backend selected transcript

### PR DESCRIPTION
**Pull request by hanars**

### CODE REVIEW

File: hail_search/hail_search_query.py [hunk 1]
```diff
@@ -618,6 +625,27 @@ class VariantHailTableQuery(BaseHailTableQuery):
...
+
+        main_transcript = ht.sorted_transcript_consequences.first()
+        if has_gene_transcripts:
+            matched_transcript = ht.gene_transcripts.first()
+            if has_allowed_transcripts:
+                allowed_transcript_ids = hl.set(ht.allowed_transcripts.map(lambda t: t.transcript_id))
+                matched_transcript = hl.or_else(
+                    ht.gene_transcripts.find(lambda t: allowed_transcript_ids.contains(t.transcript_id)),
+                    matched_transcript,
+                )
```
Comment (ShifaSZ) [inline hail_search/hail_search_query.py:641]: Moving `matched_transcript = ht.gene_transcripts.first()` into an `else` branch is cleaner.

File: hail_search/hail_search_query.py [hunk 2]
```diff
@@ -618,6 +625,27 @@ class VariantHailTableQuery(BaseHailTableQuery):
         },
     }
 
+    @staticmethod
+    def _selected_main_transcript_expr(ht):
+        has_gene_transcripts = 'gene_transcripts' in ht.row
```
Comment (ShifaSZ) [inline hail_search/hail_search_query.py:630]: The `gene_transcripts` is always annotated with either null or non-null values. So, is the `'gene_transcripts' in ht.row`  always `True`?

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: hail_search/hail_search_query.py (hunk 1 above)
Comment (hanars) [reply hail_search/hail_search_query.py:641]: I don't see how? moving it into an ekse block would mean defining it twice, we need it here and in the else statement.  leaving it where it is means we can conditionally reuse it the way we are now

File: hail_search/hail_search_query.py (hunk 2 above)
Comment (hanars) [reply hail_search/hail_search_query.py:630]: its not always annotated, the annotation is in a method thats only conditionally called

**Review by ShifaSZ — APPROVED** [review]

**Review by bpblanken — APPROVED** [review]

**Pull request merged**